app/src/main/java/ar/com/wolox/android/cookbook/notifications/NotificationFragment.kt

```diff
@@ -2,50 +2,66 @@ package ar.com.wolox.android.cookbook.notifications
 
 import android.app.PendingIntent
 import android.content.Intent
-import android.graphics.BitmapFactory
+import android.os.Build
+import android.provider.Settings
+import android.support.annotation.RequiresApi
 import android.support.v4.app.NotificationCompat
+import android.view.View
 import ar.com.wolox.android.cookbook.R
 import ar.com.wolox.android.cookbook.facebooklogin.FacebookLoginRecipeActivity
 import ar.com.wolox.android.cookbook.googlelogin.GoogleLoginRecipeActivity
-import ar.com.wolox.android.cookbook.notifications.helper.NotificationHelper
+import ar.com.wolox.android.cookbook.notifications.helper.NotificationFactory
 import ar.com.wolox.android.cookbook.notifications.model.BasicNotification
+import ar.com.wolox.android.cookbook.notifications.model.InboxNotification
 import ar.com.wolox.android.cookbook.notifications.model.NotificationAction
 import ar.com.wolox.android.cookbook.notifications.model.PictureExpandableNotification
 import ar.com.wolox.android.cookbook.notifications.model.TextExpandableNotification
 import ar.com.wolox.wolmo.core.fragment.WolmoFragment
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
 import kotlinx.android.synthetic.main.fragment_notification.*
+import javax.inject.Inject
 
 class NotificationFragment : WolmoFragment<BasePresenter<Any>>() {
 
+    private lateinit var generalChannelId: String
+
+    @Inject
+    lateinit var mNotificationFactory: NotificationFactory
+
     override fun layout(): Int = R.layout.fragment_notification
 
-    override fun init() {}
+    override fun init() {
+        generalChannelId = getString(R.string.notifications_general_channel_id)
+
+        vSettingsChannel.visibility = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) View.VISIBLE else View.GONE
+    }
 
     override fun setListeners() {
         vSimpleNotificationBtn.setOnClickListener { showBasicNotification() }
         vExpandableTextNotificationBtn.setOnClickListener { showTextExpandableNotification() }
         vBigPictureNotificationBtn.setOnClickListener { showPictureNotification() }
         vActionsNotificationBtn.setOnClickListener { showBasicNotificationWithActions() }
-    }
+        vInboxNotificationBtn.setOnClickListener { showInboxNotification() }
 
-    override fun onBackPressed(): Boolean {
-        activity!!.finish()
-        return true
+        vSettingsChannel.setOnClickListener {
```

Comment:
I believe it would be better to do:

```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
    vSettingsChannel.setOnClickListener {
        openChannelSettings()
    }
}
```